docs/synchronization.md

```diff
@@ -147,8 +151,29 @@ Examples:
 1. [Step level semaphore](https://github.com/argoproj/argo-workflows/blob/main/examples/synchronization-tmpl-level.yaml)
 1. [Step level mutex](https://github.com/argoproj/argo-workflows/blob/main/examples/synchronization-mutex-tmpl-level.yaml)
 
-### Other Parallelism support
+## Queuing
+
+When a Workflow cannot take a lock it will be placed into a ordered queue.
+
+Workflows can have a `priority` set in their specification.
+The queue is first ordered by priority, with a higher priority number being placed before a lower priority number.
+The queue is then ordered by `CreationTimestamp` of the Workflow; older Workflows will be ordered before newer workflows.
+
+Workflows are only be allowed to take a lock if they are at the front of the queue for that lock.
+
+## Workflow level parallelism
+
+You can restrict parallelism within a workflow using `parallelism` within a workflow or template.
+This only restricts total concurrent executions of steps or tasks within the same workflow.
```

Comment:
Yea I was thinking about that too, but I'm not sure that's something a user would assume changes in a Workflow vs WorkflowTemplate -- generally, it shouldn't, as that's how templates work in Argo.
Beginners may think that initially, but once you write enough workflows, you realize that's an overcomplication and there's no special case logic like that and nowhere do the docs suggest that either. Saying this in a separate sentence may also make you think there are other cases that are different (there aren't).

My alternative suggestion in that case was:
```suggestion
You can use `parallelism` within a workflow or template to restrict the total concurrent executions of steps or tasks within the same workflow.
```
But that reads a bit repetitively.

There's also no mention of WorkflowTemplate in any of the variations, so if you substitute the word "workflow" with "workflowtemplate", you may very well _still_ think it applies to all child workflows.
So I actually don't like this one or the original. The first suggestion I made makes no implication whatsoever as such.

If anything, it might be better to use the first suggestion plus a `!!! Note`, but that kind of note seems better in the WorkflowTemplate docs actually

Reply:
I'm still not preferring any of this, I think my version may be pendantic but it is unambiguous.
* I cannot find a way to misinterpret it as "there being other cases".
* It would be clear to a beginner.
* I want the documentation for a field to be all in one place, not needing to discover this field has more documentation in the WFT page.